rust/src/lib/route.rs

```diff
@@ -323,6 +331,17 @@ impl RouteEntry {
                 self.next_hop_addr = Some(new_via);
             }
         }
+        if let Some(src) = self.source.as_ref() {
+            let new_src = format!("{}", src.parse::<std::net::IpAddr>()?);
+            if src != &new_src {
+                log::warn!(
+                    "Route source address {} sanitized to {}",
```

Comment:
It is not a warning, the uncompressed IPv6 address is still legal. Use `log::info`.